runtime/resource.go

```diff
@@ -0,0 +1,113 @@
+// Package runtime is a service runtime manager
+package runtime
+
+import "fmt"
+
+const (
+	TypeNamespace     = "namespace"
+	TypeNetworkPolicy = "networkpolicy"
+	TypeService       = "service"
+)
+
+// Resource represents any resource handled by runtime
+type Resource interface {
+	ID() string
+	Type() string
```

Comment:
This will save us from doing reflection / trial-and-error assertions to figure out what each resource is, but it _does_ mean that this needs to be set. The `New*` methods of course set it, but this will pose problems for anyone putting these structs together by hand.

Reply:
I think this is fine, as long as the struct, e.g. Service always returns a const doesn't depend on New() to set a field on the struct. The reason being that there will be tens or hundreds of places that runtime.Service gets initialised in go-micro / micro and this will prevent having to change all those clients to use the New func.

Reply:
Sorry this was a non-issue